docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/google-tag-manager-custom-template/v3-tags/tag-template-guide/index.md

```diff
@@ -168,6 +168,82 @@ Using the **Context Entities** table allows you to attach [custom context entiti
 
 You can also choose to [set the True Timestamp](/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/tracking-events/index.md#setting-the-true-timestamp) with this field. The format must be UNIX time in milliseconds.
 
+### Using Plugins
+
+[Plugins](/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/plugins/index.md) provide an easy way to distribute custom behavior for the tracker.
+Each Plugin can be loaded from external URLs or provided inline within GTM, and can optionally take some custom configuration.
+A Plugin may add new methods to the tracker that can be called via [Custom Commands](#custom-commands).
+
+![The Load Plugins table opened with one external plugin configured.](images/load_plugins.png)
+
+Plugins are configured by using the **Load Plugins** table.
+Each row in the table takes three values and loads a single plugin.
+Plugins are loaded in order, and processed before the configured **Tag Type** configuration is executed -- so you can load a plugin in the same Tag that uses its functionality via Custom Commands.
+The plugins will remain configured in the tracker and be accessible to later Tags.
+
+#### External Plugins
+
+You can load external plugins by entering their URL in the **Plugin URL** field.
+The tracker will try to load plugins each time they are requested.
+This is usually acceptable as the browser will have cached the first request; if you don't want this behavior we suggest creating an empty **Custom Command** Tag with no commands that loads the plugins; you can make this Tag fire [once per page](https://support.google.com/tagmanager/answer/6279951) and add it as a [Setup Tag](https://support.google.com/tagmanager/answer/6238868) to any other Tags that use its commands.
```

Comment:
So if you have a plugin that adds API methods to the tracker, e.g. it adds `trackAwesomeEvent`, and you want to call that method, you need a custom command to call `trackAwesomeEvent`, and you need to load the plugin that makes `trackAwesomeEvent` available; the intuitive place to configure that plugin is the Custom Command Tag that calls `trackAwesomeEvent`. If `trackeAwesomeEvent` happens multiple times on the page, every time your Tag with Custom Commands fires, the tracker will try to load the plugin every time, because the plugins are loaded every time the Tag executes. If the plugin isn't written in such a way that it's aware it could be installed multiple times, it may initialize itself every time it loads, which could put it into an unknown state (e.g. attach duplicate listeners, duplicate entities, etc). If the file wasn't cached for some reason, it will also be requested multiple times per pageview whenever the Tag fires.